ipalloc.c

```diff
@@ -1,4 +1,3 @@
-#include "alloc.h"
```

Comment:
One can't add a dependency to a header file, all targets that use that header need that dependency still specified.

Reply:
I'm suggesting for e.g. `ipalloc.o`:

```diff
@@ -771,7 +771,7 @@ compile ip.c fmt.h scan.h ip.h
        ./compile ip.c

 ipalloc.o: \
-compile ipalloc.c alloc.h gen_allocdefs.h ip.h ipalloc.h ip.h \
+compile ipalloc.c gen_allocdefs.h ip.h ipalloc.h ip.h \
 gen_alloc.h
        ./compile ipalloc.c
```

Reply:
You're right, of course. I wonder how a new file using `gen_allocdefs.h` would know to include `alloc.h` in its build rules, so I'm wondering whether we should leave the includes as they were.

Reply:
gen_allocdefs.h _needs_ to include alloc.h, as it uses functions from there. Otherwise one could use it without including alloc.h first and then the symbols would be missing. We _could_ leave the includes also in the users of ```gen_allocdefs.h```, but none of the ```alloc.h``` symbols is used in those files.

The worst that could happen with the change is that one uses ```gen_allocdefs.h``` without adding the additional dependency in ```Makefile```, so necessary rebuilds on changes in ```alloc.h``` would be missing. Both are unlikely events IMHO, and would not harm a release build in any way.

Reply:
Agree, I can accept that risk too. (The underlying problem is that we’ll need to do something for #28. That can wait until whenever we get to it.)